### PR TITLE
Engg GUI: split fitting tab into its own MVP widget

### LINKED ISSUE #16618: Engg GUI: extract the fitting tab into a widget of its own
As part of #15371.

This is purely about moving code around so that the fitting tab becomes a standalone widget.
The widget should then be usable from the Engg GUI.
A bit of communication from the Engg GUI to the "calibration tab widget" might be required to grab the current calibration parameters, as those are defined by the calibration tab.
Later on the widget should be more flexible/reusable so that it can be reused for the new multi-peak/whole pattern fitting tab.

### PR DESCRIPTION
**Pull request by FedeMPouzols**
This just moves code around. There should be no effective changes in functionality.

**To test:**

Check that the fitting tab still works as it should. Example files and test steps from previous PRs can be used (I can explain what files are required and the steps to to calibrate->focus->fit if you need to generate files from scratch):
- #16281 
- #16211 
- #15991

Fixes #16618.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp [hunk 1]
```diff
@@ -0,0 +1,1126 @@
...
+                         singlePeaksWs + " or " + g_focusedFittingWSName +
+                         " workspace found.\n";
+    m_view->showStatus("Error while fitting peaks");
+    return;
+  }
+
+  try {
+    auto focusedPeaksWS =
+        ADS.retrieveWS<MatrixWorkspace>(g_focusedFittingWSName);
+    auto focusedData = ALCHelper::curveDataFromWs(focusedPeaksWS);
```
Comment (raquelalvarezbanos) [inline MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp]: I know this has not been introduced in this PR, but just wanted to say that, although this helper was initially written for the muon ALC interface, we should rename it and move it to a shared location if it is going to be useful for other interfaces.

File: MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp [hunk 2]
```diff
@@ -0,0 +1,1126 @@
...
+                          "Please try again");
+    }
+  } else {
+    m_view->userWarning("Invalid Run Number",
+                        "The specfied range of run number "
+                        "entered is invalid. Please try again");
+  }
+}
+
+void EnggDiffFittingPresenter::processStart() {}
```
Comment (raquelalvarezbanos) [inline MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp]: Is this method intended to be used in the future?

Comment (FedeMPouzols) [inline MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp]: Yes. For example, at present you only get a preview plot after you run `Fit`. But a preview of the raw data before fitting is useful. The tab should show something on the preview plot if a run number is already selected (either in the current session or saved from previous sessions), without waiting for the user to run Fit. Functionality like that should be controlled/triggered from here.

File: MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp (hunk 1 above)
Comment (FedeMPouzols) [inline MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp]: That could be done as part of the next reorganization of this code (move stuff to presenter and add tests) or as a separate issue/ticket. @shahrozahmed1.

**Comment by raquelalvarezbanos** [comment]
Code looks good to me:

-  The original MPV pattern is now applied at a lower level: the "Fitting" tab has its own view and presenter (functionality is not completely split but current code will be re-organized in future issues).
- The "Fitting" presenter is able to get the calibration settings it needs to perform the fitting from the main view via the interface `IEnggDiffractionCalibration`.
- This PR is just the first step in a big refactoring in which the different tabs will use their own MVP pattern

:shipit:

**Comment by shahrozahmed1** [comment]
`note:` while using this as a base branch I realised that the previous settings are no longer saved upon closure of the window

**Comment by FedeMPouzols** [comment]
@shahrozahmed1: good catch. The shutdown call that saves the settings for the fitting tab was missing. It should be working well now. I merged the latest master in to sort the merge conflict out.

**Pull request merged**